rosidl_generator_c/src/string_functions.c

```diff
@@ -43,7 +44,7 @@ rosidl_generator_c__String__fini(rosidl_generator_c__String * str)
     str->size = 0;
     str->capacity = 0;
   } else {
-    /* ensure that data, szie and and capacity values are consistent */
+    /* ensure that data, size, and capacity values are consistent */
     assert(0 == str->size);
     assert(0 == str->capacity);
```

Comment:
If `str->data` evaluates to false, then it points to a null pointer and does not have a "null" character allocated to it. Therefore it is in the fini'd state where both size and capacity are 0.